File: application/app/src/main/java/com/yahyeet/boardbook/model/handler/EntityHandler.java
package com.yahyeet.boardbook.model.handler;

import com.yahyeet.boardbook.model.entity.AbstractEntity;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public interface EntityHandler<T extends AbstractEntity> {

	CompletableFuture<T> find(String id);

	CompletableFuture<T> save(T entity);

	CompletableFuture<List<T>> all();
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/handler/EntityHandler.kt
package com.yahyeet.boardbook.model.handler

import com.yahyeet.boardbook.model.entity.AbstractEntity
import java.util.concurrent.CompletableFuture

interface EntityHandler<T : AbstractEntity> {

    fun find(id: String): CompletableFuture<T>

    fun save(entity: T): CompletableFuture<T>

    fun all(): CompletableFuture<List<T>>
}
